tests_e2e/playwright/navigator.py

```diff
@@ -0,0 +1,448 @@
+
+        self.page.wait_for_load_state("networkidle")
+        self.page.get_by_role("button", name="Next").click()
+
+        # let the page load
+        self.page.wait_for_load_state("networkidle")
+
+    def login_password(self) -> None:
+        """Go to a nebari deployment, login via Username/Password, and start
+        a new server.
+        """
+        logger.debug(">>> Sign in via Username/Password")
+        self.page.goto(self.nebari_url)
+        expect(self.page).to_have_url(re.compile(f"{self.nebari_url}*"))
+
+        self.page.get_by_role("button", name="Sign in with Keycloak").click()
+        self.page.get_by_label("Username").fill(self.username)
+        self.page.get_by_label("Password").click()
+        self.page.get_by_label("Password").fill(self.password)
+        self.page.get_by_role("button", name="Sign In").click()
+
+        # let the page load
+        self.page.wait_for_load_state("networkidle")
+
+    def start_server(self) -> None:
+        """Start a nebari server. There are several different web interfaces
+        possible in this process depending on if you already have a server
+        running or not. In order for this to work, we look for html elements,
+        wait because the page may need to load, and if they don't appear, we
+        check for another server start option.
+        """
+        # if server is not yet running
+        start_locator = self.page.get_by_role("button", name="Start My Server")
+        try:
+            start_locator.wait_for(timeout=3000, state="attached")
+            start_locator.click()
+
+            # I'm not sure why this is no longer needed. Leaving it here for
+            # future reference.
+            # # your server is not yet running, would like to start it?
+            # self.page.get_by_role("button", name="Launch server").click()
+
+            # select instance type (this will fail if this instance type is not
+            # available)
+            self.page.locator(f"#profile-item-{self.instance_name}").click()
+            self.page.get_by_role("button", name="Start").click()
+
+            # # wait for server to spinup
+            # server_message = self.page.get_by_text(
+            #     f"Server ready at /user/{self.username}/",
+            #     exact=True,
+            # )
+            # server_message.wait_for(
+            #     timeout=self.wait_for_server_spinup,
+            #     state="attached",
+            # )
+
+        except Exception:
+            # if the server is already running
+            start_locator = self.page.get_by_role(
+                "button",
+                name="My Server",
+                exact=True,
+            )
+            start_locator.wait_for(timeout=3000, state="attached")
+            start_locator.click()
+
+        # wait for server spinup
+        self.page.wait_for_url(
+            f"{self.nebari_url}user/{self.username}/*",
```

Comment:
This is missing a slash?

```suggestion
            f"{self.nebari_url}/user/{self.username}/*",
```

Reply:
The cypress url in GH Action has a test, I was unifying both tests so this slash is not necessary. A better solution would be to handle either case in the init.